File: src/december/December1.java
package december;

import model.Elf;
import util.Input;

import java.util.*;

public class December1 extends December {
	private final Input input;
	public December1(Input input) {
		this.input = input;
	}
	@Override
	public long star1() {
		return input()
			.stream()
			.mapToLong(Elf::getCalories)
			.max().orElse(0);
	}

	@Override
	public long star2() {
		return input()
			.stream()
			.map(Elf::getCalories)
			.sorted(Comparator.reverseOrder())
			.limit(3)
			.mapToLong(i -> i)
			.sum();
	}

	private List<Elf> input() {
		return Arrays
			.stream(input.get(1).split("\n\n"))
			.map(b -> {
				Elf e= new Elf();
				e.fillBag(b.split("\n"));
				return e;
			})
			.toList();
	}
}

File: src/december/December1.kt
package december

import util.Input

class December1 : December<Int, Int> {
    override fun star1(): Int = input().max()

    override fun star2(): Int = input().sortedDescending().take(3).sum()

    private fun input(): List<Int> =
        Input.get(1)
        .split("\n\n")
        .map { it
            .split("\n")
            .dropLastWhile { it.isEmpty() }
            .sumOf { s -> s.toInt() }
        }
        .toList()
}